### PR TITLE
feat: upgrade to PostCSS 8

### PR DESCRIPTION
**Pull request by armano2**
This PR upgrades the module to PostCSS 8 along with a few other changes:

- Upgraded ESLint to V7
- Upgraded postcss-tape to V6
- Change `postcss-value-parser` to `postcss-values-parser`
- Changed the engines since PostCSS supports only Node.js 10 and up
- Travis has been changed to GhActions

see csstools/postcss-preset-env#191

ci: https://github.com/armano2/postcss-media-fn/actions/runs/1431998088

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ h1 {
 Add [Media()] to your build tool:
 
 ```bash
-npm install postcss-media-fn --save-dev
+npm install postcss postcss-media-fn --save-dev
```
Comment (Antonio-Laguna) [inline README.md]: I think this makes sense the way it was given it's consistent with the rest of the repos. See https://github.com/csstools/postcss-nesting  or https://github.com/csstools/postcss-advanced-variables

```suggestion
npm install postcss-media-fn --save-dev
```

File: index.js [hunk 2]
```diff
@@ -1,22 +1,30 @@
-// tooling
-const postcss = require('postcss');
-const parser = require('postcss-value-parser');
+const postcss = require("postcss");
```
Comment (Antonio-Laguna) [inline index.js]: This should be single-quoted to be consistent.

```suggestion
const postcss = require('postcss');
```

File: index.js [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
+const {parse} = require('postcss-values-parser');
```
Comment (Antonio-Laguna) [inline index.js]: Let's also add spaces around here so it's consistent with the rest of the plugins.

```suggestion
const { parse } = require('postcss-values-parser');
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  push:
+    branches: master
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 16.x, 17.x]
```
Comment (Antonio-Laguna) [inline .github/workflows/ci.yaml:13]: Given we're not supporting 10.x with this change, I assume we should get rid of it.

**Review by Antonio-Laguna — CHANGES_REQUESTED** [review]
Hey @armano2!

Thanks a lot for the time you put into this PR and for getting this up to date with PostCSS 8! Really appreciate the fact you did create a GH Workflow to get this testable with GitHub actions.

I've taken a look and all seems sensible and I've just left some minor styling comments.

Nice work!

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (armano2) [reply .github/workflows/ci.yaml:13]: my mistake, i updated package.json to

```
  "engines": {
    "node": "^10 || ^12 || >=14"
  },
```

File: index.js (hunk 3 above)
Comment (armano2) [reply index.js]: applied

File: index.js (hunk 2 above)
Comment (armano2) [reply index.js]: applied

**Comment by armano2** [comment]
@Antonio-Laguna requested changes applied

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (Antonio-Laguna) [reply .github/workflows/ci.yaml:13]: Why not just `"node": ">=10"` ?

Comment (armano2) [reply .github/workflows/ci.yaml:13]: node 11, and 13 is not supported

https://github.com/postcss/postcss/blob/7cccfac2a5d0b143186df86dd72591ec0ce6135b/package.json#L5-L7

File: README.md [hunk 5]
```diff
@@ -127,10 +127,10 @@ grunt.initConfig({
...
 ```
 
-[cli-url]: https://travis-ci.org/jonathantneal/postcss-media-fn
-[cli-img]: https://img.shields.io/travis/jonathantneal/postcss-media-fn.svg
-[css-img]: https://jonathantneal.github.io/css-db/badge/media-expressions.svg
-[css-url]: https://jonathantneal.github.io/css-db/#media-expressions
+[cli-url]: https://github.com/csstools/postcss-media-fn/actions/workflows/ci.yaml/badge.svg
+[cli-img]: https://github.com/csstools/postcss-media-fn/actions/workflows/ci.yaml
+[css-img]: https://cssdb.org/badge/media-expressions.svg
+[css-url]: https://cssdb.org/#media-expressions
```
Comment (Antonio-Laguna) [inline README.md]: Given this is but an RFC these two links are 404ing. Could you get rid of these?

**Review by Antonio-Laguna — APPROVED** [review]

**Comment by Antonio-Laguna** [comment]
Great work @armano2 !

**Pull request merged**